cli/packages/cmd/user.go

```diff
@@ -0,0 +1,212 @@
+			util.HandleError(err, "[infisical user switch]: Prompt error")
+		}
+
+		//write to config file
+		configFile, err := util.GetConfigFile()
+		if err != nil {
+			util.HandleError(err, "[infisical user switch]: Unable to get config file")
+		}
+
+		configFile.LoggedInUserEmail = profile
+
+		//set logged in user domain
+		ok := util.ConfigContainsEmail(configFile.LoggedInUsers, profile)
+
+		if !ok {
+			//profile not in loggedInUsers
+			configFile.LoggedInUsers = append(configFile.LoggedInUsers, models.LoggedInUser{
+				Email:  profile,
+				Domain: config.INFISICAL_URL,
+			})
+			//set logged in user domain
+			configFile.LoggedInUserDomain = config.INFISICAL_URL
+
+		} else {
+			//exists, set logged in user domain
+			for _, v := range configFile.LoggedInUsers {
+				if profile == v.Email {
+					configFile.LoggedInUserDomain = v.Domain
+					break
+				}
+			}
+		}
+
+		err = util.WriteConfigFile(&configFile)
+		if err != nil {
+			util.HandleError(err, "")
+		}
+	},
+}
+
+var domainCmd = &cobra.Command{
+	Use:                   "domain",
+	Short:                 "Used to update the domain of an Infisical profile",
+	DisableFlagsInUseLine: true,
+	Example:               "infisical user domain",
+	Args:                  cobra.ExactArgs(0),
+	PreRun: func(cmd *cobra.Command, args []string) {
+		util.RequireLogin()
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		//prompt for profiles selection
+		loggedInProfiles, err := getLoggedInUsers()
+		if err != nil {
+			util.HandleError(err, "[infisical user domain]: Unable to get logged Profiles")
+		}
+
+		//prompt user
+		profile, err := LoggedInUsersPrompt(loggedInProfiles)
+		if err != nil {
+			util.HandleError(err, "[infisical user domain]: Prompt error")
+		}
+
+		//prompt to update domain
+		domain, err := NewDomainPrompt()
+		if err != nil {
+			util.HandleError(err, "[infisical user domain]: Prompt error")
+		}
+
+		//write to config file
+		configFile, err := util.GetConfigFile()
+		if err != nil {
+			util.HandleError(err, "[infisical user]: Unable to get config file")
+		}
+
+		//check if profile in logged in profiles
+
+		//if not add new profile loggedInUsers
+		//else update profile from loggedinUsers list
+		ok := util.ConfigContainsEmail(configFile.LoggedInUsers, profile)
+		if !ok {
+			configFile.LoggedInUsers = append(configFile.LoggedInUsers, models.LoggedInUser{
+				Email:  profile,
+				Domain: domain,
+			})
+		} else {
+			//exists, set logged in user domain
+			for _, v := range configFile.LoggedInUsers {
+				if profile == v.Email {
+					v.Domain = domain
+					break
+				}
+			}
+
+		}
+		//check if current loggedinuser is selected profile
+		//if yes set current domain to changed domain
+		if configFile.LoggedInUserEmail == profile {
+			configFile.LoggedInUserDomain = domain
+		}
+
+	},
+}
+
+func init() {
+	userCmd.AddCommand(domainCmd)
+	userCmd.AddCommand(switchCmd)
+	rootCmd.AddCommand(userCmd)
+}
+
+func getLoggedInUsers() ([]string, error) {
```

Comment:
i think it'd be great if you can write quick function descriptions in 1-2 sentences for all functions you create. It would help future contributors to quickly understand

Reply:
Yeah, thanks